modules/payloads/singles/cmd/windows/powershell_reverse_tcp.rb

```diff
@@ -5,45 +5,45 @@
 require 'rex/powershell'
 
 module MetasploitModule
-
   CachedSize = :dynamic
 
   include Msf::Payload::Single
   include Rex::Powershell::Command
   include Msf::Payload::Windows::Powershell
 
   def initialize(info = {})
-    super(merge_info(info,
-      'Name'          => 'Windows Interactive Powershell Session, Reverse TCP',
-      'Description'   => 'Interacts with a powershell session on an established socket connection',
-      'Author'        =>
-        [
+    super(
+      merge_info(
+        info,
+        'Name' => 'Windows Interactive Powershell Session, Reverse TCP',
+        'Description' => 'Interacts with a powershell session on an established socket connection',
+        'Author' => [
           'Ben Turner', # benpturner
           'Dave Hardy' # davehardy20
         ],
-      'References'    =>
-        [
-          ['URL', 'https://www.nettitude.co.uk/interactive-powershell-session-via-metasploit/']
+        'References' => [
+          ['URL', 'https://blog.nettitude.com/uk/interactive-powershell-session-via-metasploit/']
         ],
-      'License'       => MSF_LICENSE,
-      'Platform'      => 'windows',
-      'Arch'          => ARCH_CMD,
-      'Handler'       => Msf::Handler::ReverseTcpSsl,
-      'Session'       => Msf::Sessions::PowerShell,
-      'RequiredCmd'   => 'generic',
-      'Payload'       =>
-        {
-          'Offsets' => { },
+        'License' => MSF_LICENSE,
+        'Platform' => 'windows',
+        'Arch' => ARCH_CMD,
+        'Handler' => Msf::Handler::ReverseTcp,
```

Comment:
It looks like the same process should be done for `payload/windows/x64/powershell_reverse_tcp` and `payload/windows/powershell_reverse_tcp` payloads, which seems to be broken now. Since their handler is still `Msf::Handler::ReverseTcpSsl`, they are expecting an SSL connection, which is now disabled when calling `generate_powershell_code("Reverse")`.

Maybe these payloads should also be split in two modules, with and without SSL?

Reply:
Well spotted, this is fixed now!